### PR TITLE
ptask: init at 1.0.0

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Motivation for this change

- Adds a simple GTK GUI for managing taskwarrior tasks

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/ptask/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
+  name = "ptask-1.0.0";
+
+  src = fetchurl {
+    url = "http://wpitchoune.net/ptask/files/${name}.tar.gz";
+    sha256 = "13nirr7b29bv3w2zc8zxphhmc9ayhs61i11jl4819nabk7vy1kdq";
+  };
+
+  buildInputs = [ pkgconfig makeWrapper gtk3 json_c ];
```
Comment (fpletz) [inline pkgs/applications/misc/ptask/default.nix]: `pkgconfig` and `makeWrapper` are `nativeBuildInputs`.

**Review by fpletz — CHANGES_REQUESTED** [review]
You forgot to add the package to `pkgs/top-level/all-packages.nix`.

File: pkgs/applications/misc/ptask/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl, pkgconfig, makeWrapper, gtk3, json_c, taskwarrior }:
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/ptask/default.nix]: You have taskwarrior here in the input of the function, but not in any buildinputs.

File: pkgs/applications/misc/ptask/tw-version.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+--- a/src/tw.c	2017-02-21 22:56:19.289037608 +0100
++++ b/src/tw.c	2017-02-21 22:57:04.726814256 +0100
+@@ -116,7 +116,7 @@
+ 	    || !strcmp(ver, "2.3.0")
+ 	    || !strcmp(ver, "2.4.0")
+ 	    || !strcmp(ver, "2.4.1")
+-	    || !strcmp(ver, "2.5.0"))
++	    || !strcmp(ver, "2.5.1"))
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/ptask/tw-version.patch]: I can see why we need this patch, but I would really opt for contributing this upstream as well... I'm not sure why there is still the `2.5.0` version in here... Maybe both work? Have you tested it?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Overall it looks good, besides my two annotations.

File: pkgs/applications/misc/ptask/default.nix (hunk 2 above)
Comment (spacefrogg) [inline pkgs/applications/misc/ptask/default.nix]: It appears in the call to `wrapProgram` as it is a run-time dependency.

File: pkgs/applications/misc/ptask/tw-version.patch (hunk 3 above)
Comment (spacefrogg) [inline pkgs/applications/misc/ptask/tw-version.patch]: I haven't had the time yet to contribute it upstream. `ptask` has a `--force` switch to circumvent this switch, but this obviously degrades usability. It does work, though, as 2.5.1 is a maintenance release. I would leave the patch until it is fixed upstream. As `ptask` is meant to be a graphical interface, it does not make a lot of sense to force the user to go back to the terminal to make it work.

**Comment by spacefrogg** [comment]
@fpletz, @matthiasbeyer  Are we good to merge, now? I think I addressed all concerns.

**Review by matthiasbeyer — APPROVED** [review]

**Pull request merged**